scripts/zones/Alzadaal_Undersea_Ruins/Zone.lua

```diff
@@ -136,6 +140,16 @@ function onRegionEnter(player,region)
                 player:startEvent(1, tpz.besieged.getMercenaryRank(player))
             end
         end,
+        [24] = function (x)
+            if transformationsProgress == 2 then
+                player:startEvent(2)
+            end
+        end,
+        [25] = function (x)
+            if transformationsProgress <= 2 then
```

Comment:
Should probably be transformationsProgress == 3 😉

Reply:
this will require further testing as you have mentioned these CS are not optional. For example, wiggo's capture only shows the 2nd CS on that area and not the 1st one.

Reply:
It looks like the order is always Caedarva Mire exit -> Mt Zhayolm exit. Currently both of these can be skipped the way char vars are being checked at the QM and NM.